src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java

```diff
@@ -56,15 +56,19 @@ private void deleteObjectsInBucket(String dataBucket) {
     try {
       Iterable<Blob> blobs = storage.list(dataBucket).iterateAll();
       for (Blob blob : blobs) {
-        boolean suceeded = storage.delete(blob.getBlobId());
-        if (suceeded) {
-          log.info("Deleted object {}", blob);
-        } else {
-          log.warn("Could not delete object {}", blob);
+        try {
+          boolean suceeded = storage.delete(blob.getBlobId());
+          if (suceeded) {
+            log.info("Deleted object {}", blob);
+          } else {
+            log.warn("Could not delete object {}", blob);
+          }
+        } catch (Exception e) {
+          log.warn("Error deleting object {} in bucket {}", blob, dataBucket, e);
         }
       }
     } catch (Exception e) {
-      log.warn("Error deleting objects in bucket {}", dataBucket, e);
+      log.warn("Error fetching objects in bucket {} for deletion", dataBucket, e);
```

Comment:
This case isn't tested, so you need to have your mock throw an exception on `list()` (or `iterateAll()`) and then just check that there are no more interactions on `storage`.